functime/feature_extraction/tsfresh.py

```diff
@@ -198,13 +198,27 @@ def autocorrelation(x: TIME_SERIES_T, n_lags: int) -> FLOAT_EXPR:
     elif n_lags == 0:
         return 1.0
 
-    return (
-        x.shift(periods=-n_lags)
-        .drop_nulls()
-        .sub(x.mean())
-        .dot(x.shift(periods=n_lags).drop_nulls().sub(x.mean()))
-        .truediv((x.len() - n_lags).mul(x.var(ddof=0)))
-    )
+    if isinstance(x, pl.Series):
+        y = x.drop_nulls()
+        y_mean = y.mean()
+        return (
+            (
```

Comment:
@MathieuCayssol  In the series case, since we know the length of the series, it might be faster to do y.head() or y.tail to remove the nulls. Can you give it a try? Drop null is super fast but head/tail might just be faster as it is now guaranteed that y's null only show up in the head/tail.